### PR TITLE
Add ha circular progress

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Convert `<paper-spinner-lite>` to LitElement and make available as `<ha-spinner>`

Replaced all places where we use `paper-spinner-lite` currently. Still a bunch of places we use `<paper-spinner>` that should be updated. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-spinner.ts [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+  private _coolingDown = false;
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+
+    if (!changedProps.has("active")) {
+      return;
+    }
+
+    const oldActive = changedProps.get("active") as boolean;
```
Comment (bramkragten) [inline src/components/ha-spinner.ts:39]: Boolean or undefined for the first update

**Review by bramkragten — COMMENTED** [review]
Any reason not to use mwc/mdc circulair progress?

**Comment by balloob** [comment]
I didn't realize they had one! I will adapt.

File: src/components/ha-spinner.ts (hunk 1 above)
Comment (balloob) [reply src/components/ha-spinner.ts:39]: I initialize the property with `false`, so it is never undefined

File: yarn.lock [hunk 2]
```diff
@@ -1394,6 +1394,18 @@
     "@material/typography" "7.0.0-canary.d92d8c93e.0"
     tslib "^1.9.3"
 
+"@material/circular-progress@^6.0.0":
```
Comment (balloob) [inline yarn.lock:1397]: Ugh this should be v7

**Pull request merged**